sources/osg/VertexArrayObject.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+var MACROUTILS = require('osg/Utils');
+var Object = require('osg/Object');
+var GLObject = require('osg/GLObject');
+var Timer = require('osg/Timer');
+
+/**
+ * VertexArrayObject "Bundles" multipel vertex / normal / ... buffer bind
+ * in one webgl call
+ * @class VertexArrayObject
+ */
+var VertexArrayObject = function() {
+    GLObject.call(this);
+    // maybe could inherit from Object
+    this._instanceID = Object.getInstanceID();
+    this._vaoObject = undefined;
+    this.dirty();
+};
+
+// static cache of glBuffers flagged for deletion, which will actually
+// be deleted in the correct GL context.
+VertexArrayObject._sDeletedGLVertexArrayObjectCache = new window.Map();
+
+// static method to delete Program
+VertexArrayObject.deleteGLVertexArrayObject = function(gl, buffer) {
+    if (!VertexArrayObject._sDeletedGLVertexArrayObjectCache.has(gl))
+        VertexArrayObject._sDeletedGLVertexArrayObjectCache.set(gl, []);
+    VertexArrayObject._sDeletedGLVertexArrayObjectCache.get(gl).push(buffer);
+};
+
+// static method to flush all the cached glPrograms which need to be deleted in the GL context specified
+VertexArrayObject.flushDeletedGLVertexArrayObjects = function(gl, availableTime) {
```

Comment:
You need to add the call here https://github.com/cedricpinson/osgjs/blob/master/sources/osgViewer/View.js#L318. Otherwise memory is not being released.

Reply:
Fixed